### DISCUSSION TITLE
Four Coupled PDE Set Up

### DISCUSSION THREAD

**Discussion by beantsja** [usnistgov/fipy#1029] [Q&A]
I tried a coupled PDE with two equations and it worked fine, and now am trying to set up four coupled PDEs, solving the diffusion equation with the concentrations of A,B,C, and D in 1D. 

Here is a set up I have, where I will add the source terms later. Is this how I set the coupled PDE? Because with this way, the result of the solver is not behaving as I expected after adding the source terms. 

The boundary condition is no flux for all sides and all species. 

```python
m = Grid1D(nx=10, Lx=10.)
A = CellVariable(mesh=m, hasOld=True, value=A_initial)
B = CellVariable(mesh=m, hasOld=True, value=B_initial)
C = CellVariable(mesh=m, hasOld=True, value=C_initial)
D = CellVariable(mesh=m, hasOld=True, value=D_initial)


eq0 = TransientTerm(var=A) == DiffusionTerm(coeff=DA, var=A) +...(source terms)
eq1 = TransientTerm(var=B) ==  DiffusionTerm(coeff=DB, var=B) +...(source terms)
eq2 = TransientTerm(var=C) ==  DiffusionTerm(coeff=DC, var=C) +...(source terms)
eq3 = TransientTerm(var=D) ==  DiffusionTerm(coeff=DD, var=D) +...(source terms)

eqn = eq0 & eq1 & eq2 & eq3

for t in range(10):
    A.updateOld()
    B.updateOld()
    C.updateOld()
    D.updateOld()
    eqn.solve(dt=1e-1)
    print(H)
```

Also, what is the role of updateOld()? I haven't seen this from other examples. If there is any other thing that I should consider, please let me know as well. Thank you.

**Comment by wd15** [comment]
Looking at the code above, it seems that the equations don't really depend on each other in an implicit way unless they are coupled via an  `ImplicitSourceTerm`. I'm assuming that there must be some explicit coupling in the source terms. Putting them all in one matrix doesn't give you any benefit in this case. Maybe solve them separately and check that the solution is consistent?

Regarding `updateOld`, `CellVariable` keeps the value of the variable from the previous step as the `TransientTerm` requires the previous value when using non-linear iterations within a time step (although you're not doing this). `updateOld` just triggers the `CellVariable` to update the old value with the current value. If you didn't call `updateOld` then the solution wouldn't evolve.
